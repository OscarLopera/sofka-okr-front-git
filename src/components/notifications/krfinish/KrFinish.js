import React, { useState, useEffect } from 'react';
import Modal from 'react-modal';
import '../notificationsWindows.scss'
import axios from 'axios'



const HOST_API = "http://localhost:8080/notifications";


const KrFinish = () => {
    
    const [modalIsOpen, setModalIsOpen] = useState(false)
    const [request, setRequest] = useState({keyResult:"", name:""});

    const completedkr = () => {
        axios.get("http://localhost:8080/notifications/completedkr/jvsabvfbvfeh")
        .then(rest=>{
            setRequest({keyResult: rest.data[0], name: rest.data[1]})

        })
    }
    
    useEffect( () => {completedkr()}, [])
    
    const customStyles = {
        content: {
            top: '20%',
            left: '50%',
            right: 'auto',
            bottom: 'auto',
            marginRight: '-50%',
            width: '600px',
            transform: 'translate(-40%, -10%)',
            padding: '0',
        },
    };

    return (
        <div >
            <button onClick={() => setModalIsOpen(true)} id='buttonNotificationKr'>Terminar Kr</button>
            <Modal isOpen={modalIsOpen}
                onRequestClose={() => setModalIsOpen(false)}
                style={customStyles}>
                <header className='headerNotificationWindows' id='headerNotificationWindows'></header>
                <div className='bodyNotificationWindows' id='bodyNotificationWindows'>
                     <h1>¡{request.name}!</h1>
                    <h3>Terminaste el kr {request.keyResult}</h3> 
                    <p>"Nuestra recompensa se encuentra en el esfuerzo y no en el
                    resultado, un esfuerzo total es una víctoria completa (Mahatma Gandhi)".</p>
                </div>
                <div className='imageNotificationWindows' id='imageNotificationWindows'>
                    <img src="https://www.sofka.com.co/wp-content/uploads/2020/08/sofka-logo-gradient-white.png" alt="logo-sofka"></img>
                </div>
                <footer className='footerNotificationWindows' id='footerNotificationWindows'></footer>
            </Modal>
        </div>
    )
}

export default KrFinish;