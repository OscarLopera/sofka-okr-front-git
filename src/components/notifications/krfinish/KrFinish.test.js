import React from 'react';
import Modal from 'react-modal';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import KrFinish from './KrFinish'
import sinon from 'sinon'

configure({ adapter: new Adapter() });

it('show content of modal', () => {
    const spy = sinon.spy();
    const wrapper = shallow(<KrFinish onTest={spy}></KrFinish>)
    expect(spy.called)
    wrapper.unmount();
})

it('renders react-modal', () => {
    const wrapper = shallow(<KrFinish />);
    expect(wrapper.find(Modal)).toHaveLength(1);
});

it('opens modal when button is clicked', () => {
    const wrapper = shallow(<KrFinish />);
    expect(wrapper.find(Modal).prop('isOpen')).toBe(false);
    wrapper.find('button').simulate('click');
    expect(wrapper.find(Modal).prop('isOpen')).toBe(true);
});
