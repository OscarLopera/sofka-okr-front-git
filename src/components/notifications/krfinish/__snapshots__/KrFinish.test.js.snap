// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders childen when modal is open 1`] = `
Array [
  <header
    className="headerNotificationWindows"
    id="headerNotificationWindows"
  />,
  <div
    className="bodyNotificationWindows"
    id="bodyNotificationWindows"
  >
    <h1>
      ¡ 
      Andres Caro
      ! 
    </h1>
    <h2>
       
      Estudio de Encuesta
    </h2>
    <h3>
      Terminaste el kr 
      Realizar Encuesta a 100 personas
    </h3>
    <p>
      "Nuestra recompensa se encuentra en el esfuerzo y no en el resultado, un esfuerzo total es una víctoria completa (Mahatma Gandhi)".
    </p>
  </div>,
  <div
    className="imageNotificationWindows"
    id="imageNotificationWindows"
  >
    <img
      alt="logo-sofka"
      src="https://www.sofka.com.co/wp-content/uploads/2020/08/sofka-logo-gradient-white.png"
    />
  </div>,
  <footer
    className="footerNotificationWindows"
    id="footerNotificationWindows"
  />,
]
`;
